core/src/main/java/com/linecorp/armeria/common/AggregatedHttpRequest.java

```diff
@@ -298,4 +291,15 @@ default Locale selectLocale(Locale... supportedLocales) {
     default HttpRequest toHttpRequest() {
         return HttpRequest.of(headers(), content(), trailers());
     }
+
+    /**
+     * Converts this request into a new complete {@link HttpRequest} replacing the {@link #headers()}
+     * with the specified {@link RequestHeaders}.
+     *
+     * @return the new {@link HttpRequest} converted from this request.
+     */
+    default HttpRequest toHttpRequest(RequestHeaders headers) {
+        requireNonNull(headers, "headers");
+        return HttpRequest.of(maybeModifyContentLength(headers, content()), content(), trailers());
```

Comment:
Doesn't have to be done in this PR, just a note:

I find it surprising that `HttpRequest.of` modifies the content based on header length, I'm wondering if the behavior should be consistent with `AggregatedHttpRequest.of`

Reply:
I realized the above comment was incorrect 😅 
One question though, if the `Content-Length` is adjusted within the static constructor anyways, why do we do it one more time at `maybeModifyContentLength`?

Reply:
Thanks, updated. 😉